ponyup-init.sh

```diff
@@ -27,6 +28,17 @@ BLUE="\e[34m"
 RED="\e[31m"
 YELLOW="\e[33m"
 
+shasumCommand() {
+  if command -v sha256sum > /dev/null 2>&1; then
+    sha256sum "$@"
+  elif command -v shasum > /dev/null 2>&1; then
+    shasum --algorithm 256 "$@"
+  else 
+    printf "%bNo checksum command found.%b\n" "${RED}" "${DEFAULT}" >&2
```

Comment:
No `>&2` is needed for how we are currently doing error output.

If you want, you can open an issue to address error message not going to STDERR. I'd want that to be part of a separate PR though.

Reply:
I made it like this to ensure printing the error message, otherwise it will look like this:
<img width="913" alt="Screen Shot 2020-02-04 at 9 24 04 PM" src="https://user-images.githubusercontent.com/2499718/73805457-b378ed00-4794-11ea-84d2-c23584bb829a.png">

But with `pipefail` and redirecting to `stderr` it looks like this:
<img width="905" alt="Screen Shot 2020-02-04 at 9 25 03 PM" src="https://user-images.githubusercontent.com/2499718/73805510-e622e580-4794-11ea-891a-fa1f9669b3e8.png">

Do you know how to solve it and comply with posix?

Reply:
Not off the top of my head. Can you open an issue for that? It's different than what you are currently tackling so best to do in its own issue and PR.

Reply:
So, it's ok for this PR to have the wrong error message?

Reply:
Wait, so...

without the redirect it isn't exiting correctly?

Reply:
it's exiting, but the error message will be misleading! like the first screenshot!

Reply:
Ok, well then, the shasum call will need to be handled in a more compliant fashion that doesn't require the pipefail. I'm not sure what that would be off the top of my head. You'll need to investigate that.

Sorry, I didn't understand what the issue was until now. Apologies.

Reply:
No worries, I'll investigate it and see what can I do

Reply:
It exits as a result of the checksum failing. The exit in the function doesn't appear to be working. I'm not sure why and don't have time at the moment to look, you'll need to figure out why it isn't an address in a posix compliant fashion.

My suggestion would be near the start of the script, test for a valid shasum and set a variable to that, then use the variable rather than a function call. And you can exit at the top.

Does that make sense? I hope I was clear.

I'd say do that check and variable set up before:

`ponyup_root="${prefix}/ponyup"`

Reply:
Makes sense, that's what I was thinking too.

Reply:
Done